bridge/src/lib.rs

```diff
@@ -849,10 +910,29 @@ pub mod pallet {
 
 		/// unpause all registered domains in the storage
 		fn unpause_all_domains() {
+			// iterate over the registered domain storage first and write all of them into IsPaused
+			// storage
+			for registered_domain_pair in DestDomainIds::<T>::iter() {
+				IsPaused::<T>::insert(registered_domain_pair.0, false);
+			}
+			// iterate over all IsPaused storage to set them as false(unpaused)
 			for bridge_pair in IsPaused::<T>::iter() {
```

Comment:
It seems duplicated with above?

Reply:
it seems a bit duplicated, but my idea was that iteration over the `registered_domain` storage first and insert all domains into the `IsPaused` storage, then iteration over `IsPaused` to pause/unpause them, this will make sure to sync between `registered_domain` and `IsPaused`. Because when register a new domain A, it only exists in `registered_domain` storage, if we pause all domain at this moment, A will not be pause because it doesn't exist in `IsPaused` storage.

Reply:
Get it, thanks for explanation